bin/git-forgit

```diff
@@ -263,7 +263,7 @@ _forgit_diff() {
     opts="
         $FORGIT_FZF_DEFAULT_OPTS
         +m -0 --bind=\"enter:execute($FORGIT diff_enter {} $escaped_commits | $_forgit_enter_pager)\"
-        --preview=\"$FORGIT diff_view {} $_forgit_preview_context $escaped_commits\"
+        --preview=\"$FORGIT diff_view {} \"$_forgit_preview_context\" $escaped_commits\"
```

Comment:
Is this really the correct thing to do here? As far as I can see we're assigning the `opts` variable, so the escaped quotes will be a quote in the variable's value, which would make this line:

```
--preview="$FORGIT diff_view {} "$_forgit_preview_context" $escaped_commits"
```

So actually this does not really quote the variable in the `--preview` argument. Or am I wrong?

Reply:
I think it is. From my understanding, `fzf` evaluates the string that is passed to `--preview`, which essentially removes the outer quotes.

Reply:
But evaluating the string would also remove the inner quotes, because the first one of them closes the very first quote, and the second one opens a new quote to be closed by the very last quote. Wouldn't it?

Reply:
I see you point now and you're right. I think the correct way to escape this would be like this then:

```sh
--preview=\"$FORGIT diff_view {} \\\"$_forgit_preview_context\\\" $escaped_commits\"
```
Do you agree?
We would also have to update several other lines that make the same mistake, e.g. in `_forgit_reset_head` we do the same thing

```sh
    opts="
        $FORGIT_FZF_DEFAULT_OPTS
        -m -0
        --preview=\"$FORGIT reset_head_preview \"$rootdir\"/{}\"
        $FORGIT_RESET_HEAD_FZF_OPTS
    "
```

Reply:
Glad you're seeing it as well. You're totally right, we should fix it everywhere. The double escape you suggested seems correct. It's hardly readable, though. Would single quotes work here instead? Or would that be bad practice? It shouldn't prevent the shell variables from being evaluated because that happens when assigning `opts` already, so the `--preview` argument would have constant values in it. We could choose if we'd either replace the outer or the inner double quotes with single quotes. WDYT?